### ISSUE TITLE
SCF of 48 atoms Ni CANNOT converge

### ISSUE THREAD

**Issue by Luc1anoo** [deepmodeling/abacus-develop#3954]
### Describe the Testing Issue

[Ni.zip](https://github.com/deepmodeling/abacus-develop/files/14940812/Ni.zip)

SCF of 48 atoms Ni CANNOT converge. The STRUs is constructed by ATOMSK in .cif format and then converted into STRU format using ase-abacus (python3 cif2STRU.py), and relaxation is not been performed before SCF calculation because it also CANNOT converge(see "relax" directory).
The structures in "structure" are also constructed by ATOMSK by moving the top half atom layers to calculate the stacking fault energy. The STRU I'm testing is equal to  "STRU_0" file, which is the initial structure with perfect lattice.
I have redirected the stdout into result-3.3.2 and result-3.6.0 respectively.

version: ABACUS v3.3.2, ABACUS v3.6.0

### Task list for Issue attackers (only for developers)

- [ ] Understand the testing issue described by the developer.
- [ ] Review the specific test case, expected and actual results, and any error messages.
- [ ] Identify the root cause of the test failure or issue.
- [ ] If a possible solution is suggested, evaluate its feasibility and effectiveness.
- [ ] Implement a fix for the test failure or issue, or create a new test case if needed.
- [ ] Verify that the fix resolves the testing issue and the test case passes.
- [ ] Review and update any relevant documentation, such as test plans or user guides.
- [ ] Ensure the testing issue is resolved and close the ticket.
- [ ] Share any lessons learned or best practices with the team to prevent similar issues in the future.

**Comment by QuantumMisaka** [comment]
@Luc1anoo Can you upload your INPUT, KPT and STRU files ?

**Comment by Luc1anoo** [comment]
> @Luc1anoo Can you upload your INPUT, KPT and STRU files ?

They are all in the Ni.zip file.

**Comment by QuantumMisaka** [comment]
@Luc1anoo By changing initial magnetic moment to 1.0 and using `mixing_ndim 20` can make it converge within 200 electronic steps, I'm having more test

**Comment by QuantumMisaka** [comment]
@Luc1anoo @WHUweiqingzhou 

[Ni-conv.tar.gz](https://github.com/deepmodeling/abacus-develop/files/14969862/Ni-conv.tar.gz)

Just set initial magnetic moment to 1.0 and using `mixing_ndim 20` (with `smearing_sigma 0.008 ~ 0.010` can make SCF converge within 70 steps.

Also, for this slab-like structure, rotate the slab direction to y direction will accelerate the calculation process.

**Comment by Luc1anoo** [comment]
> @Luc1anoo @WHUweiqingzhou
> 
> [Ni-conv.tar.gz](https://github.com/deepmodeling/abacus-develop/files/14969862/Ni-conv.tar.gz)
> 
> Just set initial magnetic moment to 1.0 and using `mixing_ndim 20` (with `smearing_sigma 0.008 ~ 0.010` can make SCF converge within 70 steps.
> 
> Also, for this slab-like structure, rotate the slab direction to y direction will accelerate the calculation process.

Thanks for the reply. I have tested that the SCF can indeed converge with THIS STRU(i.e. the initial/ the "STRU_0" in the "structure" directory in the above Ni.zip). But with the NEXT STRU(i.e. the "STRU_1" in the "structure" directory), the SCF cannot converge again. I also tested setting the mixing_beta=0.1 for this STRU and it didn't work. You cna check these information in the following "TEST" file(whose INPUT is same as for STRU_0) and "beta0.1"(only set the mixing_beta from 0.4 to 0.1), both STRU is "STRU_1" in the "structure" directory.
[TEST.zip](https://github.com/deepmodeling/abacus-develop/files/14987434/TEST.zip)
[beta0.1.zip](https://github.com/deepmodeling/abacus-develop/files/14987435/beta0.1.zip)

**Comment by QuantumMisaka** [comment]
@Luc1anoo Also, if you change `calculation` to `relax`, you will found that the largest gradient after SCF converge is 0.02 and the relax job will be done in 1 ion step

**Comment by Luc1anoo** [comment]
> @Luc1anoo @WHUweiqingzhou
> 
> [Ni-conv.tar.gz](https://github.com/deepmodeling/abacus-develop/files/14969862/Ni-conv.tar.gz)
> 
> Just set initial magnetic moment to 1.0 and using `mixing_ndim 20` (with `smearing_sigma 0.008 ~ 0.010` can make SCF converge within 70 steps.
> 
> Also, for this slab-like structure, rotate the slab direction to y direction will accelerate the calculation process.

Thanks for the reply! I tested your file (re-calc) and the scf calculation seems fine. It's strange that the scf calculation can't converge when I use the same INPUT as yours on my STRUs, which is just a little different in format than yours.(These STRUs are first constructed by Atomsk in cif format and then transfered into STRU format. I use these 20 STRUs in `structure` directory to perform stacking fault calculation.) The two scf calculation files named `test1` and `test2` are as follows:
[test1.zip](https://github.com/deepmodeling/abacus-develop/files/14994405/test1.zip)
[test2.zip](https://github.com/deepmodeling/abacus-develop/files/14994404/test2.zip)
I can't see the difference between these two files in performing scf calculation, but the later just can't converge.

**Comment by Luc1anoo** [comment]
> @Luc1anoo Also, if you change `calculation` to `relax`, you will found that the largest gradient after SCF converge is 0.02 and the relax job will be done in 1 ion step

Thanks a lot! Maybe I will just relax the structure to get energy.

**Comment by WHUweiqingzhou** [comment]
@Luc1anoo,
I have already reproduced your result, see [here](https://app.bohrium.dp.tech/abacustest/?request=GET%3A%2Fapplications%2Fabacustest%2Fjobs%2Fjob-ni-abacustest-v0.3.112-f6a351). I think it is a bug.

**Comment by Luc1anoo** [comment]
I also tested the structure whose top half atom layer(the `STRU_1` in the `structure` directory in the `Ni.zip` above) has been moved and found it cannot converge, with setting the mag by element. The input and output file can be seen in the following `test3`.For comparision, the output file `result` includes the result of scf calculation of unmoved structure with setting the mag by element, which can converge, and then the moved structure, which cannot. I noticed that the `TMAG` will always oscillate in the scf calculation which cannot converge. I think the problem may relate to the Ni system.
[test3.zip](https://github.com/deepmodeling/abacus-develop/files/15102560/test3.zip)
